internal/plugin/connectors/tcp/mssql/production_options_test.go

```diff
@@ -0,0 +1,133 @@
+package mssql
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cyberark/secretless-broker/internal/plugin/connectors/tcp/mssql/types"
+	mssql "github.com/denisenkom/go-mssqldb"
+)
+
+func TestProductionReadLoginRequest(t *testing.T) {
+	// production version of ReadLoginRequest
+	var readLoginRequest types.ReadLoginRequestFunc = mssql.ReadLoginRequest
+
+	// expected login request available from net.Conn passed to ReadLogin
+	expectedLoginRequest := &mssql.LoginRequest{}
+	expectedLoginRequest.AppName = "test-app-name"
+	expectedLoginRequest.UserName = "test-user-name"
+	expectedLoginRequest.Database = "test-database"
+	expectedLoginRequest.SSPI = []uint8{}
+
+	r, w := net.Pipe()
+	go func() {
+		_ = mssql.WriteLoginRequest(w, expectedLoginRequest)
+	}()
+
+	// login request returned from ReadLoginRequest
+	actualLoginRequest, _ := readLoginRequest(r)
+
+	assert.Equal(t, actualLoginRequest, expectedLoginRequest)
+}
+
+func TestProductionWriteLoginResponse(t *testing.T) {
+	// production version of WriteLoginResponse
+	var writeLoginResponse types.WriteLoginResponseFunc = mssql.WriteLoginResponse
+
+	// expected login request available from net.Conn passed to WriteLoginResponse
+	expectedLoginResponse := &mssql.LoginResponse{}
+	expectedLoginResponse.Interface = 23
+	expectedLoginResponse.ProgName = "test-progname"
+	expectedLoginResponse.ProgVer = 01
+	expectedLoginResponse.TDSVersion = 12
+
+	r, w := net.Pipe()
+	go func() {
+		writeLoginResponse(w, expectedLoginResponse)
+	}()
+
+	// login response returned from ReadLoginResponse
+	actualLoginResponse, err := mssql.ReadLoginResponse(r)
+	assert.Nil(t, err)
+	if err != nil {
+		return
+	}
+
+	assert.Equal(t, expectedLoginResponse, actualLoginResponse)
+}
+
+func TestProductionWritePreLoginResponse(t *testing.T) {
```

Comment:
Am I being *really* anal to want to move the `PreLogin` test cases before the `Login` test cases, and have the `TestProductionReadPreLoginResponse` test case before the `TestProductionWritePreloginResponse`? (i.e. chronological order?)

Reply:
It's definitely a small change, but an important one 👍